### PR TITLE
Fix failure in IT test with external cluster

### PR DESCRIPTION
**Pull request by zhongnansu**
*Issue #, if available:*
#391 causes github workflow failure
*Description of changes:*
1. Fix failure in IT test with external cluster, basically the previous PR #391 which caused a github workflow failure
2. Add try with resource to avoid JMX connector won't close when an exception is thrown. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLIntegTestCase.java [hunk 1]
```diff
@@ -107,23 +106,25 @@ protected boolean preserveClusterUponCompletion() {
...
-        connector.close();
+        try(JMXConnector connector = JMXConnectorFactory.connect(new JMXServiceURL(serverUrl))) {
+            IProxy proxy = MBeanServerInvocationHandler.newProxyInstance(
+                    connector.getMBeanServerConnection(), new ObjectName("org.jacoco:type=Runtime"), IProxy.class,
+                    false);
+
+            Path path = Paths.get(jacocoBuildPath + "/integTest.exec");
+            Files.write(path, proxy.getExecutionData(false));
+        } catch (Exception ex) {
+            throw new RuntimeException("Failed to dump coverage: " + ex);
```
Comment (dai-chen) [inline src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLIntegTestCase.java]: Would be better if we keep the full error stack by `throw new RuntimeException("Failed to dump coverage", ex);`

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLIntegTestCase.java [hunk 2]
```diff
@@ -107,23 +106,25 @@ protected boolean preserveClusterUponCompletion() {
     }
 
     @AfterClass
-    public static void dumpCoverage() throws IOException, MalformedObjectNameException {
+    public static void dumpCoverage() {
         // jacoco.dir is set in sqlplugin-coverage.gradle, if it doesn't exist we don't
         // want to collect coverage so we can return early
         String jacocoBuildPath = System.getProperty("jacoco.dir");
-        if (jacocoBuildPath.isEmpty()) {
+        if (jacocoBuildPath == null || jacocoBuildPath.isEmpty()) {
```
Comment (dai-chen) [inline src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLIntegTestCase.java]: np: `isNullOrEmpty` in Strings or StringUtils is cleaner.

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLIntegTestCase.java (hunk 1 above)
Comment (zhongnansu) [reply src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLIntegTestCase.java]: Agreed.

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLIntegTestCase.java (hunk 2 above)
Comment (zhongnansu) [reply src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLIntegTestCase.java]: good catch, use Strings.isNullOrEmpty(Guava) instead.

**Pull request merged**